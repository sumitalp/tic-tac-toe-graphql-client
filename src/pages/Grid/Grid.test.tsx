import * as React from "react";
import { shallow } from "enzyme";
import GridPage from ".";
import { Symbol as ISymbol } from "../../../typings/types";

describe("GridPage", () => {
    const unsubscribeMock = jest.fn();
    const defaultProps: any = {
        controllingPlayer: { symbol: ISymbol.CROSS },
        subscribeToGridUpdates: jest.fn().mockReturnValue(unsubscribeMock),
        gameUrls: {
            [ISymbol.NAUGHT]: "",
            [ISymbol.CROSS]: ""
        }
    };

    beforeEach(() => {
        jest.clearAllMocks();
    });

    it("should render", () => {
        const component = shallow(<GridPage {...defaultProps} />);

        expect(component).toMatchSnapshot();
    });
});
