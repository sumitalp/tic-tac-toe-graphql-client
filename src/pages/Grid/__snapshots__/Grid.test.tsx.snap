// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GridPage should render 1`] = `
<Component
  controllingPlayer={
    Object {
      "symbol": "CROSS",
    }
  }
  gameUrl="http://localhost/"
>
  <Grid
    controllingPlayer={
      Object {
        "symbol": "CROSS",
      }
    }
    gameUrls={
      Object {
        "CROSS": "",
        "NAUGHT": "",
      }
    }
    isDraw={false}
    subscribeToGridUpdates={[MockFunction]}
    winner={null}
  />
</Component>
`;
